src/usr/share/netcheck/networkmanagerhelper.py

```diff
@@ -0,0 +1,202 @@
+
+        return success
+
+    def get_network_ip(self, network_id):
+        """Attempts to retrieve the ip address associated with the given network id. If it
+        is unable to, it returns None."""
+        # I decided not to throw an exception here because the intended caller would end up
+        #   simply using it for flow control.
+        ip_address = None
+
+        connection = self.network_id_table[network_id]
+
+        if self._wait_for_connection(connection):
+            device = self._get_device_for_connection(connection)
+            ip_address = device.Ip4Config.AddressData[0]['address']
+        else:
+            self.logger.warning('Attempted to get IP address for network that is not connected.')
+
+        return ip_address
+
+    def network_is_ready(self, network_id):
+        """Check whether the network with the given network id is ready."""
+        connection = self.network_id_table[network_id]
+        return self._wait_for_connection(connection)
+
+    def _build_network_id_table(self):
+        """Assemble a helpful dictionary of network objects, indexed by the connection's
+        id in NetworkManager.
+        """
+        network_id_table = {}
+
+        all_connections = NetworkManager.Settings.ListConnections()
+        self._run_proxy_call(all_connections)
+
+        for connection in all_connections:
+            connection_id = connection.GetSettings()['connection']['id']
+            network_id_table[connection_id] = connection
+
+        return network_id_table
+
+    def _get_active_connection(self, network_id):
+        """Returns the active connection object associated with the given network id."""
+
+        # The active connection objects returned by NetworkManager are very short-lived and
+        #   not directly accessible from the connection object.
+
+        active_connection_list = NetworkManager.NetworkManager.ActiveConnections
+        self._run_proxy_call(active_connection_list)
+
+        for listed_active_connection in active_connection_list:
+            if listed_active_connection.Id == network_id:
+                self.logger.trace('Found active connection.')
+                return listed_active_connection
+
+        return None
+
+    def _get_connection_state(self, network_id):
+        """Returns the state of the connection with the given network id."""
+        state = NM_CONNECTION_DISCONNECTED
+
+        active_connection = self._get_active_connection(network_id)
+
+        if active_connection is None:
+            self.logger.warning('Connection disconnected.')
+
+        else:
+            if hasattr(active_connection, 'State'):
+                if active_connection.State is NetworkManager.NM_ACTIVE_CONNECTION_STATE_ACTIVATING:
+                    state = NM_CONNECTION_ACTIVATING
+                if active_connection.State is NetworkManager.NM_ACTIVE_CONNECTION_STATE_ACTIVATED:
+                    state = NM_CONNECTION_ACTIVE
```

Comment:
It seems like the term 'active' is a bit overloaded. Is there something we can do about that? I'd probably leave this constant alone and rename the _get_active_connection method.